internal/tokenstatus/token_status.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tokenstatus provides token status functions
+package tokenstatus
+
+import (
+	"sync"
+
+	"github.com/stacklok/minder/internal/db"
+)
+
+// Store stores the rate limit status of token
+type Store struct {
+	status map[key]chan struct{}
```

Comment:
> Have you considered using `xsync.MapOf`? We already use it in minder, so you wouldn't drag in a new dependency and you wouldn't have to take care of the mutex yourself.

We use `v1.5.2` which is not supported now. and even`v3` had a bug fix a few months back:

> Fix too aggressive Map/MapOf shrinking on deletion leading to out of range panic

First of all, I'd suggest that we update the version as many bugs have been fixed since `v1.5.2`. Secondly, I'm open to using `xsync.MapOf` but I'm not too sure about the stability given that a recent version fixed a out of range panic. Wdyt?

Reply:
I don't have a great answer, I'm even surprised we were using such an old version. I don't see an issue with your implementation, I think the way you used `Lock`/`Rlock` etc is already correct and given that these are used in network traffic I'm not too concerned about lock congestion being a bottleneck.

So let's leave the code as-is and let me see about upgrading `xsync.MapOf`

Reply:
I wonder if dependabot catches updates when a newer version is released i.e. `yourcoolpackage/v3`.